code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,674 @@
+		S.alpha = 0
+		sparkles += S
+		switch(i)
+			if(1 to 8)
+				S.orbit(src, 30, TRUE, 60, 36, TRUE, FALSE)
+			if(9 to 16)
+				S.orbit(src, 62, TRUE, 60, 36, TRUE, FALSE)
+			if(17 to 24)
+				S.orbit(src, 95, TRUE, 60, 36, TRUE, FALSE)
+			if(25)
+				S.pixel_y = 7
+				S.loc = get_turf(src)
+		sleep(7)
+	for(var/obj/reveal in sparkles)
+		reveal.alpha = 255
+	while(active)
+		for(var/obj/item/device/flashlight/spotlight/glow in spotlights) // The multiples reflects custom adjustments to each colors after dozens of tests
+			if(!src || QDELETED(src) || !active || !glow || QDELETED(glow))
+				return
+			if(glow.light_color == "red")
+				glow.light_color = "nw"
+				glow.light_power = glow.light_power * 1.48
+				glow.light_range = 0
+				glow.update_light()
+				continue
+			if(glow.light_color == "nw")
+				glow.light_color = "green"
+				glow.light_range = glow.range * 1.1
+				glow.light_power = glow.light_power * 2 // Any changes to power must come in pairs to neutralize it for other colors
+				glow.update_light()
+				continue
+			if(glow.light_color == "green")
+				glow.light_color = "sw"
+				glow.light_power = glow.light_power * 0.5
+				glow.light_range = 0
+				glow.update_light()
+				continue
+			if(glow.light_color == "sw")
+				glow.light_color = "purple"
+				glow.light_power = glow.light_power * 2.27
+				glow.light_range = glow.range * 1.15
+				glow.update_light()
+				continue
+			if(glow.light_color == "purple")
+				glow.light_color = "se"
+				glow.light_power = glow.light_power * 0.44
+				glow.light_range = 0
+				glow.update_light()
+				continue
+			if(glow.light_color == "se")
+				glow.light_color = "#ffff00"
+				glow.light_range = glow.range * 0.9
+				glow.update_light()
+				continue
+			if(glow.light_color == "#ffff00")
+				glow.light_color = "ne"
+				glow.light_range = 0
+				glow.update_light()
+				continue
+			if(glow.light_color == "ne")
+				glow.light_color = "red"
+				glow.light_power = glow.light_power * 0.68
+				glow.light_range = glow.range * 0.85
+				glow.update_light()
+				continue
+		if(prob(2))
+			INVOKE_ASYNC(src, .proc/hierofunk)
+		sleep(selection.song_beat)
+
+
+/obj/machinery/disco/proc/dance(var/mob/living/carbon/M) //Show your moves
+	switch(rand(0,9))
+		if(0 to 1)
+			set waitfor = 0
+			for(var/i = 1, i < 8, i++)
+				M.SpinAnimation(7,1)
+				M.setDir(pick(cardinal))
+				sleep(10)
+		if(2 to 3)
+			set waitfor = 0
+			for(var/i in 1 to 6)
+				if (!M)
+					return
+				M.SpinAnimation(7,1)
+				M.setDir(pick(cardinal))
+				for (var/x in 1 to 12)
+					sleep(1)
+					if (!M)
+						return
+					if (i<5)
+						M.pixel_y += 1
+					if (i>4)
+						M.pixel_y -= 2
+					M.setDir(turn(M.dir, 90))
+					switch (M.dir)
+						if (NORTH)
+							M.pixel_y += 3
+						if (SOUTH)
+							M.pixel_y -= 3
+						if (EAST)
+							M.pixel_x -= 3
+						if (WEST)
+							M.pixel_x += 3
+				sleep(12)
+			M.pixel_x = 0
+			M.pixel_y = 0
```

Comment:
Yes, but this is outside of the waitfor. What will happen is the inner code will run, go to sleep, then this code will run and exit the proc, then the inner code will wake up and keep changing the offsets.